sql: apply PARTITION ALL BY when using ALTER PRIMARY KEY

otan opened the pull request:
Release note (sql change): PARTITION ALL BY statements will now apply
for tables when using ALTER PRIMARY KEY.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59178)
<!-- Reviewable:end -->

otan committed:
sql: apply PARTITION ALL BY when using ALTER PRIMARY KEY

Release note (sql change): PARTITION ALL BY statements will now apply
for tables when using ALTER PRIMARY KEY.

ajstorm reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59178#-:-MRVv9-H-RuTtfVCi_b0:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajstorm

thx
